tools/stats/check_disabled_tests.py

```diff
@@ -0,0 +1,290 @@
+            except json.JSONDecodeError:
+                stats = {}
+
+            all_tests[disabled_test_id]["num_green"] += stats.get("num_green", 0)
+            all_tests[disabled_test_id]["num_red"] += stats.get("num_red", 0)
+        elif failure:
+            # As a failure, increase the failure count
+            all_tests[disabled_test_id]["num_red"] += 1
+        else:
+            all_tests[disabled_test_id]["num_green"] += 1
+
+    return all_tests
+
+
+def get_test_reports(
+    repo: str, workflow_run_id: int, workflow_run_attempt: int
+) -> Generator[Path, None, None]:
+    """
+    Gather all the test reports from S3 and GHA. It is currently not possible to guess which
+    test reports are from rerun_disabled_tests workflow because the name doesn't include the
+    test config. So, all reports will need to be downloaded and examined
+    """
+    with TemporaryDirectory() as temp_dir:
+        print("Using temporary directory:", temp_dir)
+        os.chdir(temp_dir)
+
+        artifact_paths = download_s3_artifacts(
+            "test-reports", workflow_run_id, workflow_run_attempt
+        )
+        for path in artifact_paths:
+            unzip(path)
+
+        artifact_paths = download_gha_artifacts(
+            "test-report", workflow_run_id, workflow_run_attempt
+        )
+        for path in artifact_paths:
+            unzip(path)
+
+        for report in Path(".").glob("**/*.xml"):
+            yield report
+
+
+def get_disabled_test_name(test_id: str) -> Tuple[str, str, str, str]:
+    """
+    Follow flaky bot convention here, if that changes, this will also need to be updated
+    """
+    name, classname, filename = test_id.split(SEPARATOR)
+    return f"{name} (__main__.{classname})", name, classname, filename
+
+
+def prepare_record(
+    workflow_id: int,
+    workflow_run_attempt: int,
+    name: str,
+    classname: str,
+    filename: str,
+    flaky: bool,
+    num_red: int = 0,
+    num_green: int = 0,
+) -> Tuple[Any, Dict[str, Any]]:
+    """
+    Prepare the record to save onto S3
+    """
+    key = (
+        workflow_id,
+        workflow_run_attempt,
+        name,
+        classname,
+        filename,
+    )
+
+    record = {
+        "workflow_id": workflow_id,
+        "workflow_run_attempt": workflow_run_attempt,
+        "name": name,
+        "classname": classname,
+        "filename": filename,
+        "flaky": flaky,
+        "num_green": num_green,
+        "num_red": num_red,
+    }
+
+    return key, record
+
+
+def save_results(
+    workflow_id: int,
+    workflow_run_attempt: int,
+    all_tests: Dict[str, Dict[str, int]],
+) -> None:
+    """
+    Save the result to S3, so it can go to Rockset
+    """
+    should_be_enabled_tests = {
+        name: stats
+        for name, stats in all_tests.items()
+        if "num_green" in stats
+        and stats["num_green"]
+        and "num_red" in stats
+        and stats["num_red"] == 0
+    }
+    still_flaky_tests = {
```

Comment:
i think this includes just failing tests too right

Reply:
Yup, anything that doesn't satisfy the success condition is treated as "still" flaky including:

* No `num_green` stats
* num_green == 0
* No `num_red` stats (suspicious, in case the script parses thing wrong)
* num_red > 0

```
if "num_green" in stats and stats["num_green"] and "num_red" in stats and stats["num_red"] == 0
```